README.md

```diff
@@ -217,97 +211,99 @@ The `error` return is used to signal invalid Pattern
 structure, which could be bad UTF-8 or malformed JSON 
 or leaf values which are not provided as arrays.
 
-As many Patterns as desired can be added to a Matcher. 
-The `CoreMatcher` type does not support `DeletePattern()`
-but `pruner.Matcher` does.
+As many Patterns as desired can be added to a Quamina
+instance. More than one pattern can be added with the
+same `X` identifier.
 
 The `AddPattern` call is single-threaded; if multiple
 threads call it, they will block and execute sequentially.
-
 ```go
-func (m *Matcher) MatchesForJSONEvent(event []byte) ([]X, error)
+func (q *Quamina) DeletePatterns(x X) error 
 ```
-
-The `event` argument must be a JSON object encoded in
-correct UTF-8. 
+After calling this API, no list of matches from
+`AddPattern` will include the `X` value specified
+in the argument.
 
 The `error` return value is nil unless there was an
-error in the Event JSON.
+internal failure of Quamina’s storage system.
+```go
+func (q *Quamina) MatchesForEvent(event []byte) ([]X, error)
+```
+The `error` return value is nil unless there was an
+error in the encoding of the Event.
 
 The `[]X` return slice may be empty if none of the Patterns
-match the provided Event. 
+match the provided Event.
 
-```go
-func (m *Matcher) MatchesForFields([]Field) ([]X, error)
-```
-Performs the functions of `MatchesForJSON` on an 
-Event which has been flattened into a list of `Field`
-instances.  At the moment, `CoreMatcher` only returns
-an error if the `[]Field` argument is nil. `pruner.Matcher`
-can return an error if it suffers a failure in its 
-Pattern storage.
-
-These matching calls are thread-safe. Many threads may
-be executing it concurrently, even while `AddPattern` is
-also executing.  There is a significant performance 
-penalty if there is a high rate of `AddPattern` in
-combination with matching.
+### Concurrency
 
-```go
-func NewFJ(*Matcher) Flattener
-```
-Creates a new JSON-specific Flattener.
-```go
-func (fj *FJ) Flatten([]byte event) []Field
-```
-Transforms an event, which must be JSON object
-encoded in UTF-8, into a list of `Field` instances.
+A single Quamina instance can not safely be used by
+multiple goroutines at the same time.  However, the
+underlying data structure is designed for concurrent
+access and the `Copy` API is provided to support this.
 
 ```go
-func (fj *FJ) FlattenAndMatch([]byte event) ([]X, error)
+func (q *Quamina) Copy() *Quamina
 ```
-Utility function which combines the functions of the 
-`FJ.Flatten` and `Matcher.MatchesForFields` APIs.
+
+This generates a copy of the target instance. Such
+copies may safely run in parallel in different
+goroutines executing any combination of
+`MatchesForEvent()`, `AddPattern()`, and
+`DeletePattern()` calls.  There is a significant
+performance penalty if there is a high rate of
+`AddPattern()` invocations in parallel with
+`MatchesForEvent()`.
```

Comment:
question: above you say a single instance is not thread-safe. Here you write about parallel operations (I read that as concurrent operations). At least, this is confusing as the reader might not be clear about which operations/objects are thread-safe or not :)

Reply:
well, I mean if you have a bunch of goroutines making Quamina API calls, each with a different Quamina instance, if a high proportion of those calls are AddPattern, performance will suffer.   OK, will try to clarify.